src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -164,6 +165,9 @@ private void Initialize(AnalyzerExecutor analyzerExecutor, DiagnosticQueue diagn
                 newOnAnalyzerException = (ex, analyzer, diagnostic) => addDiagnostic(diagnostic);
             }
 
+            // Assume all analyzers are non-thread safe.
```

Comment:
Even this was done for a separate change, please remove unused locals singleThreadedAnalyzerToGateMap and getAnalyzerGate.